### PR TITLE
Changed navbar to be horizontal and added open/close on vertical navbar

### PR DESCRIPTION
**Pull request by Padogen**
- new navbar
- minor fix
- Changed login placement, minor fixes in NavFavorites
- Changed width of coursebanner and fixes in Navfavorites

### CODE REVIEW

**Review by JosteinLindhom — COMMENTED** [review]
Some additional comments:
- The return button (`RedirectButton`) placement should be changed to not be ontop of the NavBar
- The banner shown on the home page when not logged in does not fill the whole width now that the sidebar no longer displayed

Take a look at indentation and excess whitespace, and remove unused imports and variables.

File: dev/src/components/NavBar.tsx [hunk 1]
```diff
@@ -1,49 +1,56 @@
-import React from "react"
-import { useAppState } from "../overmind"
-import { Link } from "react-router-dom"
-import NavBarFooter from "./navbar/NavBarFooter"
-import NavBarCourse from "./navbar/NavBarCourse"
-import { isEnrolled, isVisible } from "../Helpers"
-
+import React from "react";
+import { useActions, useAppState } from "../overmind";
+import { Link } from 'react-router-dom'
```
Comment (meling) [inline dev/src/components/NavBar.tsx:3]: Also inconsistent use of `;` and two different quote types `"` and `'`. As mentioned in PR #613, you can install the ESLint plugin to automagically fix/format the code to remove the `;` at least.

File: dev/src/components/NavBar.tsx [hunk 2]
```diff
@@ -1,49 +1,56 @@
...
-                        QuickFeed
-                    </Link>
-                </li>
-                {!state.isLoggedIn &&
-                    <li>
-                        <a href="/auth/github" style={{ textAlign: "center", paddingTop: "15px" }}>
-                            Sign in with <i className="fa fa-2x fa-github align-middle ml-2" id="github" />
+        <nav className="navbar navbar-expand-lg" style={{backgroundColor: "#222", color: "#d4d4d4"}} id="main" >
+            {!state.showFavorites &&
+            <a className="navbar-brand" style={{marginLeft: "30px", fontWeight: "bold"}}>
```
Comment (meling) [inline dev/src/components/NavBar.tsx:23]: Missing indentations here. Please use the ESLint plugin to automagically format the code.

File: dev/src/components/NavBar.tsx [hunk 3]
```diff
@@ -1,49 +1,56 @@
...
+                <Link to="/" style={{ fontWeight: "bold", fontSize: "30px", color: "#d4d4d4"}}>
+                    QuickFeed
+                </Link>
+            </a>
+            }     
+            {!state.isLoggedIn &&
+            <div className="navbar-collapse ml-auto ">
```
Comment (meling) [inline dev/src/components/NavBar.tsx:30]: Remove blank after `auto`

File: dev/src/components/NavBar.tsx [hunk 4]
```diff
@@ -1,49 +1,56 @@
...
+                <ul className="ms-auto ml-auto list-unstyled" >
```
Comment (meling) [inline dev/src/components/NavBar.tsx:31]: Remove blank before the `>`

File: dev/src/components/NavBar.tsx [hunk 5]
```diff
@@ -1,49 +1,56 @@
...
-                        <Link to="/courses" className="Sidebar-items-link">
-                            View all courses
-                        </Link>
-                    </li>}
-                <NavBarFooter key="foot" />
+                </ul>  
+            </div>
+            } 
+            {state.isLoggedIn &&
+            <ul className="mr-auto me-auto list-unstyled" >
```
Comment (meling) [inline dev/src/components/NavBar.tsx:41]: Remove the blank before the `>`

File: dev/src/components/NavBar.tsx [hunk 6]
```diff
@@ -1,49 +1,56 @@
...
+                    <NavFavorites/>
+                }
+            </div>
         </nav>
     )
-
 }
-
-export default NavBar
+export default NavBar
```
Comment (meling) [inline dev/src/components/NavBar.tsx:56]: Missing newline

File: dev/src/components/NavFavorites.tsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import React, { Component } from "react";
+import { useAppState,  useActions } from "../overmind";
+import { Link } from 'react-router-dom'
+import NavBarFooter from "./navbar/NavBarFooter";
```
Comment (meling) [inline dev/src/components/NavFavorites.tsx:4]: Unused import

File: dev/src/components/navbar/NavBarUser.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                        <AboutButton/>
+                        <AdminButton/>
+                        <LoginButton/>
+                    </ul>
+                </li>
+            </ul>
+        </div> 
+    )
+}
+export default NavBarUser
```
Comment (meling) [inline dev/src/components/navbar/NavBarUser.tsx:71]: missing newline

File: dev/src/components/navbar/NavBarUser.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        return (
+            <li>
+                <a href="/auth/github" style={{ textAlign: "center", paddingTop: "15px" }}>
+                    <i className="fa fa-2x fa-github" id="github"></i>
+                </a>
+            </li>
+        )
+    }
+    return (
+        <div className="navbar-collapse ml-auto " id="main_nav">
```
Comment (meling) [inline dev/src/components/navbar/NavBarUser.tsx:55]: remove blank after `auto`

File: dev/src/components/navbar/NavBarUser.tsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+            <ul className="navbar-nav ml-auto">
+                <li className="nav-item dropdown ml-auto " >
```
Comment (meling) [inline dev/src/components/navbar/NavBarUser.tsx:57]: remove blank after `auto`

File: dev/styles/styles.css [hunk 11]
```diff
@@ -323,3 +328,26 @@ input {
...
 }
+
+
+.dropdown-menu-center {
+  right: auto;
+  left: 50%;
+  -webkit-transform: translate(-50%, 0);
+  -o-transform: translate(-50%, 0);
+  transform: translate(-50%, 0);
+
```
Comment (meling) [inline dev/styles/styles.css:339]: Remove blank

File: dev/styles/styles.css [hunk 12]
```diff
@@ -323,3 +328,26 @@ input {
   box-shadow: rgb(0, 0, 0) 0px 0px 0px inset,
     rgba(0, 0, 0, 0.2) 5px 0px 0px inset;
 }
+
+
```
Comment (meling) [inline dev/styles/styles.css:332]: Remove blank

File: dev/styles/styles.css [hunk 13]
```diff
@@ -323,3 +328,26 @@ input {
...
+}
+
+.banner{
+  margin-left: -350px;
+
+}
+
+@media all and (min-width: 992px) {
+    .navbar .nav-item .dropdown-menu{ display: none; }
+    .navbar .nav-item:hover .nav-link{   }
```
Comment (meling) [inline dev/styles/styles.css:349]: This line gives a warning: `Do not use empty ruleset`

**Comment by meling** [comment]
Reminder to Resolve conversations that you have fixed.

PS: I've merged master into this PR branch, so remember to pull to your working copy, so you will have a similar `.vscode/settings.json` as I do. Hence, it should be able to format the code when you save a file. Please let us know on discord if you still have problems formatting the code. If so, you may need to install `clang-format`. See [here](https://github.com/quickfeed/quickfeed#contributing) for some background on `clang-format`. Installing it may be different on your system though, but it should be available [here](https://llvm.org/builds/). Although I'm not sure if this will ensure it is available in your PATH etc. Please let us know if you have issues related to this, so that we can update the docs if there are problems.

Specifically, I would like to know if `clang-format` is necessary for formatting Typescript code. I know it is required to format `.proto` files, but don't actually think it is needed for Typescript and Javascript.

**Review by meling — APPROVED** [review]
Looks good to me!

**Comment by meling** [comment]
@JosteinLindhom @Padogen @Synnems is this PR still relevant or was it covered by #626? If so, we should close this one.

**Comment by JosteinLindhom** [comment]
> @JosteinLindhom @Padogen @Synnems is this PR still relevant or was it covered by #626? If so, we should close this one.

Looks to me like everything in this PR was included, and expanded upon, in #626.

**Comment by meling** [comment]
Closing as it was already merged as part of #626

**Pull request closed without merging**